dll/cpl/sysdm/smbios.c

```diff
@@ -430,6 +465,26 @@ BOOL TrimNonPrintable(PCHAR DmiString)
     return FALSE;
 }
 
+static
+BOOL TrimNonPrintableProd(PCHAR DmiString)
+{
+    PCHAR c;
+
+    if (!DmiString)
+    {
+        return FALSE;
+    }
+
+    /* Special handling for HP with broken revision */
+    c = strstr(DmiString, "(\xFF\xFF");
```

Comment:
Oh they really have a NULL-terminated string with FF bytes??

Reply:
Yes this happens in some hardware setups / SMBIOS dumps :)

Reply:
Why does it need to be a special case? Isn't this already handled by TrimNonPrintable?

Reply:
It didn't worked as expected for some reason, already forgot why... it's about this case:

![image](https://user-images.githubusercontent.com/578406/78185877-6f445900-7474-11ea-8585-fc6034d77464.png)

Reply:
I would have expected that the TrimNonPrintable would parse the full-device name string from the beginning, then reach the opening parenthesis (still OK), then reach the character 0xFF (displayed with "ya" in the screencapture), see it's invalid and NULL-terminates the string there?

Reply:
Yes that's correct. The idea behind this additional function is to change return value to `FALSE` as if the string was not trimmed. Preprocessed string is passed to `TrimNonPrintable` and it returns `FALSE`. If we pass the string as is, it would return `TRUE`, and based on this value later the logic decides to use board strings instead of system strings, but they are more usable in this exact case. That's why additional function was introduced.

Reply:
ok then you should defintively add a comment in the code explaining that.

Reply:
Added a comment ;)